tests/test_interface.py

```diff
@@ -98,4 +98,12 @@ def test_hyperparam_entrypoint():
     func = utils._load_hyperparam_func("kalman.default")
     expected = 1
     result = func(None, None)
-    assert result == expected
\ No newline at end of file
+    assert result == expected
+
+
+def test_negative_axis():
+    t = np.arange(3)
+    x = np.ones((2, 3, 2))
+    axis = -2
+    dx = dxdt(x, t, kind='finite_difference', axis=axis, k=1)
+    assert x.shape == dx.shape
```

Comment:
Hey Yash, can you make sure that this actually differentiates the correct axis, not just that it returns the correct shape?  Similar to how we did it in the pysindy test, or how it's done [in this test in this repo](https://github.com/andgoldschmidt/derivative/blob/2bfeb89b41b744e98565f4fb12aa8f6c028b505e/tests/test_interface.py#L37C5-L40C1)